packages/std/src/math/uint128.rs

```diff
@@ -726,6 +730,18 @@ mod tests {
         assert_eq!(a, Uint128::from(30u32));
     }
 
+    #[test]
+    fn uint128_pow_works() {
+        assert_eq!(Uint128::from(2u32).pow(2), Uint128::from(4u32));
+        assert_eq!(Uint128::from(2u32).pow(10), Uint128::from(1024u32));
+    }
+
+    #[test]
+    #[should_panic]
+    fn uint128_pow_overflow_panics() {
+        Uint128(u128::MAX).pow(2u32);
```

Comment:
```suggestion
        Uint128::MAX.pow(2u32);
```

Reply:
Ah, it was bit of copy-paste. I'll fix it in entire file then.